Model/UserInterface.php

```diff
@@ -227,4 +227,74 @@ public function addRole($role);
      * @return static
      */
     public function removeRole($role);
+
+    /**
+     * Checks whether the user's account has expired.
+     *
+     * Internally, if this method returns false, the authentication system
+     * will throw an AccountExpiredException and prevent login.
+     *
+     * @return bool true if the user's account is non expired, false otherwise
+     *
+     * @see AccountExpiredException
+     */
+    public function isAccountNonExpired();
```

Comment:
- 1 for having these methods in the FOSUserBundle interface if we don't do anything useful with them (i.e. we should only have `isEnabled`)

Reply:
Same as https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2815/files/3fd0e9789ba08602b5873a164cfe5e61a8ebfd68#diff-fead1d2e8961c005678d0ae0758f1bac

If we remove this method, some people can get in trouble ;-)
This method is used in `FOS\UserBundle\Security\UserChecker::checkPreAuth()`

Reply:
That's true as long as compatibility is granted. When dropping sf4 support, maybe, this should be removed and users should, for example, implement own `UserChecker` as decorator of this if they want to rely on that field. I know that removing things from interfaces is always seen as a bad thing but here we're introducing something just to avoid breaking the logic of the framework.

Reply:
Hi @DonCallisto , I don't understand why introducing account locking in "FOS user" is such a shame? Does this bundle not manage user account? perhaps this is a missing feature from the bundle and it could permit to be fully compliant with next major symfony version and even avoid BC break...

Reply:
@yonisolo see https://symfony.com/blog/new-in-symfony-4-1-deprecated-the-advanceduserinterface

Reply:
Yes i saw that deprecation and the link to create a custom user checker but preauth check for account status could be implemented as an interface to provide compatibility and with a basic usage which could be overridden later by the user for its own business logic. Presently, if FOSUserBundle remove this functionality, most user will be lost between the Symfony security component and the FOsUser User management, please don't break this chain...

Reply:
@yonisolo No one proposed to remove `UserChecker` that's basically the "link" you're talking about. As @stof says, `isAccountNonExpired` (both on `UserChecker` and `UserInterface`) should not be part of this bundle as are not FOSUserBundle concept (IIUC) but the `UserChecker` as well as `UserInterface` will still be part of this bundle (otherwise it will be pretty useless)

Reply:
ah ok, my mistakes, sorry!! I understand better, thank you!!